src/components/OrdersWidget/OrderRow.tsx

```diff
@@ -62,32 +55,21 @@ function displayTokenSymbolOrLink(token: TokenDetails): React.ReactNode | string
   return displayName
 }
 
-function calculatePrice(numeratorString?: string | null, denominatorString?: string | null): string {
-  let price
-  if (numeratorString && denominatorString) {
-    const numerator = new BigNumber(numeratorString)
-    const denominator = new BigNumber(denominatorString)
-
-    price = formatPrice(numerator, denominator)
-  }
-
-  return price || 'N/A'
-}
-
 interface OrderDetailsProps extends Pick<Props, 'order' | 'pending'> {
   buyToken: TokenDetails
   sellToken: TokenDetails
 }
 
 const OrderDetails: React.FC<OrderDetailsProps> = ({ buyToken, sellToken, order }) => {
   const [price, priceInverse] = useMemo((): string[] => {
-    const numeratorString = formatAmountFull(order.priceNumerator, buyToken.decimals, false)
-    const denominatorString = formatAmountFull(order.priceDenominator, sellToken.decimals, false)
-    const priceFmt = calculatePrice(numeratorString, denominatorString)
-    const priceInverseFmt = calculatePrice(denominatorString, numeratorString)
-
-    return [priceFmt, priceInverseFmt]
-  }, [buyToken, order.priceDenominator, order.priceNumerator, sellToken])
+    const price = calculatePrice({
+      buyToken: { amount: order.priceNumerator, decimals: buyToken.decimals },
+      sellToken: { amount: order.priceDenominator, decimals: sellToken.decimals },
+    })
+    const priceInverse = new BigNumber(1).div(price)
```

Comment:
is price always `BN`? if not then this will throw occasionally

Reply:
Is price BN or BigNumber? If BN then maybe `price.toString(10)` is necessary

Reply:
Scratch that, it's BigNumber, all good

Reply:
Price is always BigNumber